object-storage-gcp/src/main/java/io/micronaut/objectstorage/googlecloud/GoogleCloudStorageEntry.java

```diff
@@ -53,4 +55,16 @@ public InputStream getInputStream() {
     public Blob getNativeEntry() {
         return blob;
     }
+
+    @NonNull
+    @Override
+    public Map<String, String> getMetadata() {
+        return blob.getMetadata();
```

Comment:
Is the map returned here immutable? If not wrap in `Collections.unmodifiableMap`